python/common/org/python/java/Function.java

```diff
@@ -269,8 +269,8 @@ public static MatchType parameterMatch(java.lang.Class<?> from_type, java.lang.C
 
         java.lang.reflect.Method method = null;
         java.lang.StringBuilder signature = new java.lang.StringBuilder();
-        java.lang.Class<?>[] arg_types = new java.lang.Class<?>[args.length];
-        int n_args = args.length;
+        int n_args = (args == null) ? 0 : args.length;
+        java.lang.Class<?>[] arg_types = new java.lang.Class<?>[n_args];
```

Comment:
This may be the place where the performance hit is actually taking place (or, at least, somewhere else you can potentially optimise); every 0-arg call is creating an array of size 0.

Reply:
Done!